terminus/generators/opendrive_generator.py

```diff
@@ -0,0 +1,133 @@
+            {% set distances = generator._get_distances(points) %}
+            {% set total_length = generator._road_length(distances) %}
+            {% set angles = generator._yaws(points) %}
+            <road name="{{model.name}}" length="{{total_length}}" id="{{segment_id}}">
+              <type s="0.0000000000000000e+00" type="town"/>
+              <planView>
+              {% for i in range(generator._collection_size(points) - 1)    %}
+                <geometry s="{{generator._road_length(distances[:i])}}" x="{{points[i].x}}" y="{{points[i].y}}" hdg="{{angles[i]}}" length="{{distances[i]}}">
+                  <line/>
+                </geometry>
+              {% endfor %}
+              </planView>
+              <elevationProfile>
+                  <elevation s="0.0000000000000000e+00" a="0.0000000000000000e+00" b="0.0000000000000000e+00" c="0.0000000000000000e+00" d="0.0000000000000000e+00"/>
+              </elevationProfile>
+              <lateralProfile>
+              </lateralProfile>
+              <lanes>
+                  <laneSection s="0.0000000000000000e+00">
+                      <center>
+                          <lane id="0" type="driving" level= "0">
+                              <link>
+                              </link>
+                              <roadMark sOffset="0.0000000000000000e+00" type="none" weight="standard" color="standard" width="1.3000000000000000e-01"/>
+                          </lane>
+                      </center>
+                      <left>
+                          <lane id="1" type="driving" level= "0">
+                              <link>
+                              </link>
+                              <roadMark sOffset="0.0000000000000000e+00" type="solid" weight="standard" color="standard" width="1.3000000000000000e-01"/>
+                              <width sOffset="0.0000000000000000e+00" a="{{model.get_width()/2.0}}" b="0.0000000000000000e+00" c="0.0000000000000000e+00" d="0.0000000000000000e+00"/>
+                          </lane>
+                      </left>
+                      <right>
+                          <lane id="-1" type="driving" level= "0">
+                              <link>
+                              </link>
+                              <roadMark sOffset="0.0000000000000000e+00" type="solid" weight="standard" color="standard" width="1.3000000000000000e-01"/>
+                              <width sOffset="0.0000000000000000e+00" a="{{model.get_width()/2.0}}" b="0.0000000000000000e+00" c="0.0000000000000000e+00" d="0.0000000000000000e+00"/>
+                          </lane>
+                      </right>
+                  </laneSection>
+              </lanes>
+              <objects>
+              </objects>
+              <signals>
+              </signals>
+            </road>"""
+
+    def trunk_template(self):
+        return self.street_template()
+
+    def _distance(self, pA, pB):
```

Comment:
We are using our own `Point` implementation now, so we should implement this in that class. Can you move this to it so we can do `point1.distance_to(point2)`?